### PR TITLE
Global search

### PR DESCRIPTION
**Pull request by bhattabhi013**
Implemented global search functionality

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -23,6 +24,10 @@ const routes: Routes = [
     canActivate: [HomeGuard],
     loadChildren: () => import('./pages/home/home.module').then( m => m.HomePageModule)
   },
+  {
+    path: 'global-search',
```
Comment (vishwanath1004) [inline src/app/app-routing.module.ts]: path should be search

File: src/app/components/project-card/project-card.component.html [hunk 2]
```diff
@@ -6,10 +6,10 @@
         <label class="project-name ">{{ projectName }}</label>
```
Comment (vishwanath1004) [inline src/app/components/project-card/project-card.component.html:6]: First letter of Project name should be capital.

File: src/app/components/project-card/project-card.component.html [hunk 3]
```diff
@@ -6,10 +6,10 @@
...
         </ion-col>
         <ion-col size="4" size-sm="4" size-md="4" size-lg="4" class="center-container">
-          <p [ngClass]="projectStatus === 'Started' ? 'in-progress' : 'not-started '">{{ projectStatus }}</p>
+          <ion-label text-wrap [ngClass]="projectStatus === 'Started' ? 'in-progress' : 'not-started '">{{ projectStatus }}</ion-label>
```
Comment (vishwanath1004) [inline src/app/components/project-card/project-card.component.html]: notStarted display as Not started, define the constants and match the status

File: src/app/components/project-card/project-card.component.html [hunk 4]
```diff
@@ -6,10 +6,10 @@
...
         </ion-col>
         <ion-col size="1" size-sm="1" size-md="1" size-lg="1" class="center-container">
-          <p  [ngClass]="{ 'completed': projectStatus === 'Completed', 'in-progress': projectStatus != 'Completed' }">{{ taskCount }}</p>
+          <ion-label  [ngClass]="{ 'completed': projectStatus === 'Completed', 'in-progress': projectStatus != 'Completed' }">{{ taskCount }}</ion-label>
```
Comment (vishwanath1004) [inline src/app/components/project-card/project-card.component.html]: where is the css completed class defined ?

File: src/app/components/project-card/project-card.component.scss [hunk 5]
```diff
@@ -1,18 +1,20 @@
 .in-progress {
-  padding: 10px;
+  padding: 1px;
   color: var(--ion-color-success);
```
Comment (vishwanath1004) [inline src/app/components/project-card/project-card.component.scss]: #ffa500 use this color code for in progress  status

File: src/app/components/project-card/project-card.component.scss [hunk 6]
```diff
@@ -1,18 +1,20 @@
...
-  border-radius: 20px;
+  // border-radius: 20px;
   box-shadow: none;
+  width: 100%; /* label take up the full width of the column */
   }
   
   .not-started {
-    padding: 10px;
+    padding: 1px;
     color: var(--ion-color-danger);
```
Comment (vishwanath1004) [inline src/app/components/project-card/project-card.component.scss]: #4f4b48 use this color code for in progress status

Comment (vishwanath1004) [reply src/app/components/project-card/project-card.component.scss]: make one generic class 
   .status{
box-shadow: none;
    width: 20%;
    height: 50%;
    width: 100%;
padding: 1px;
}

.not-started {
color: #4f4b48 
}

.in-progress {
color :  #ffa500 
}
do like this

File: src/app/components/project-card/project-card.component.scss [hunk 7]
```diff
@@ -54,5 +56,9 @@ ion-card-content{
 }
 
 ion-icon{
```
Comment (vishwanath1004) [inline src/app/components/project-card/project-card.component.scss]: don't write css for tags that may affect in other pages.

File: src/app/pages/global-search/global-search.module.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { NgModule } from '@angular/core';
```
Comment (vishwanath1004) [inline src/app/pages/global-search/global-search.module.ts]: this should be a component not module.

File: src/app/pages/global-search/global-search.page.ts [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    const config = {
+      url: dynamicUrl,
+    };
+
+    await this.http.setHeader();
+    this.http.get(config).subscribe((data) => {
+      this.isLoading = false;
+      if (data) {
+        if(data.result == 0){
+          this.toast.showToast('No Data found', 'success');
```
Comment (vishwanath1004) [inline src/app/pages/global-search/global-search.page.ts]: No data found should be translated

File: src/assets/i18n/en.json [hunk 10]
```diff
@@ -61,6 +61,9 @@
         "SETTINGS" : "Settings",
         "LOGOUT" : "Logout"
     },
+    "GLOBAL_SEARCH" :{
+        "TITLE" : "Global Search"
```
Comment (vishwanath1004) [inline src/assets/i18n/en.json]: Change the title Search project

File: src/app/components/project-search/project-search.component.ts [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+  templateUrl: './project-search.component.html',
+  styleUrls: ['./project-search.component.scss'],
+})
+export class ProjectSearchComponent  implements OnInit {
+
+  searchResults: any[] = [];
+  currentPage = 1;
+  isLoading = false;
+  debounceTimer: any;
+  limit = 2;
```
Comment (vishwanath1004) [inline src/app/components/project-search/project-search.component.ts]: make this limit as 25

File: src/app/components/project-search/project-search.component.ts [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+    const dynamicUrl = urlConstants.API_URLS.GLOBAL_SEARCH(this.currentPage, this.limit, this.searchTerm);
+    const config = {
+      url: dynamicUrl,
+    };
+
+    await this.http.setHeader();
+    this.http.get(config).subscribe((data) => {
+      this.isLoading = false;
+      if (data) {
+        if(data.result == 0){
```
Comment (vishwanath1004) [inline src/app/components/project-search/project-search.component.ts:61]: what is this condition?

Comment (bhattabhi013) [reply src/app/components/project-search/project-search.component.ts:61]: it generates the toast in infinite scroll and informs the user that no data is available

**Pull request merged**